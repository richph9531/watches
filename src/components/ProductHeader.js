import React from 'react';
import PropTypes from 'prop-types';
import classes from '../styles/ProductHeader.module.css';

const ProductHeader = props => (
    <div>
      <h1 className={classes.PageTitle}>{props.data.title}</h1>
      <p className={classes.PageDescription}>{props.data.description}</p>
    </div>
);

ProductHeader.propTypes = {
  data: PropTypes.object,
  title: PropTypes.string,
  description: PropTypes.string,
};

export default ProductHeader;
