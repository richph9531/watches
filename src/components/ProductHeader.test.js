import React from 'react';

import { render } from '@testing-library/react';
import ProductHeader from './ProductHeader';

test('Page header renders correctly', () => {
  const data = {
    title: 'Page header Title',
    description: 'Page header description',
  };
  const renderedElement = render(
  <ProductHeader
    data={data}
  />,
  );
  expect(renderedElement.baseElement).toMatchSnapshot();
});

test('Can render without title data', () => {
  const data = {
    description: 'Page header description',
  };
  const renderedElement = render(
  <ProductHeader
    data={data}
  />,
  );
  expect(renderedElement.baseElement).toMatchSnapshot();
});

test('Can render without description data', () => {
  const data = {
    title: 'Page header Title',
  };
  const renderedElement = render(
  <ProductHeader
    data={data}
  />,
  );
  expect(renderedElement.baseElement).toMatchSnapshot();
});

test('Can render without data', () => {
  const renderedElement = render(
  <ProductHeader
    data={{}}
  />,
  );
  expect(renderedElement.baseElement).toMatchSnapshot();
});
