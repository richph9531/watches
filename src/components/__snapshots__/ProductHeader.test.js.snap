// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can render without data 1`] = `
<body>
  <div>
    <div>
      <h1
        class="PageTitle"
      />
      <p
        class="PageDescription"
      />
    </div>
  </div>
</body>
`;

exports[`Can render without description data 1`] = `
<body>
  <div>
    <div>
      <h1
        class="PageTitle"
      >
        Page header Title
      </h1>
      <p
        class="PageDescription"
      />
    </div>
  </div>
</body>
`;

exports[`Can render without title data 1`] = `
<body>
  <div>
    <div>
      <h1
        class="PageTitle"
      />
      <p
        class="PageDescription"
      >
        Page header description
      </p>
    </div>
  </div>
</body>
`;

exports[`Page header renders correctly 1`] = `
<body>
  <div>
    <div>
      <h1
        class="PageTitle"
      >
        Page header Title
      </h1>
      <p
        class="PageDescription"
      >
        Page header description
      </p>
    </div>
  </div>
</body>
`;
